dotcom-rendering/src/web/components/FetchCommentCounts.importable.tsx

```diff
@@ -0,0 +1,173 @@
+	return `https://api.nextgen.guardianapps.co.uk/discussion/comment-counts.json?shortUrls=${discussionIds.join(
+		',',
+	)}`;
+}
+
+/**
+ * @description
+ * Converts the count number into a formatted string and parses the stringified
+ * format
+ *
+ * E.g.:
+ *   - The 'long' version of 12345 is 12,345
+ *   - The 'short' version of 12345 is 12k
+ */
+function enhanceCounts(counts: RawCountType[]): EnhancedCountType[] {
+	const markers = extractMarkers();
+	return counts.map((count) => {
+		const { long, short } = formatCount(count.count);
+		// We don't get format in the api response so look it up in the array
+		// of DOM markers
+		const format = markers.find(
+			(marker) => marker.discussionId === count.id,
+		)?.format as ArticleFormat; // We cast because we're sure we will find it
+		return {
+			id: count.id,
+			long,
+			short,
+			format,
+		};
+	});
+}
+
+/**
+ * @description
+ * Takes the long and short version of each count and generates the html
+ * for showing the count on the page
+ */
+function renderCounts(counts: EnhancedCountType[]): RenderedCountType[] {
+	return counts.map((count) => {
+		const html = renderToString(
+			<CardCommentCount
+				format={count.format}
+				short={count.short}
+				long={count.long}
+			/>,
+		);
+		return {
+			id: count.id,
+			html,
+			short: count.short,
+		};
+	});
+}
+
+/**
+ * @description
+ * Takes html and inserts it into the relevant Card
+ *
+ */
+function insertCount({
+	id,
+	html,
+	short,
+}: {
+	id: string;
+	html: string;
+	short: string;
+}) {
+	const countContainers = document.querySelectorAll<HTMLElement>(
+		`[data-discussionid="${id}"]`,
+	);
+	countContainers.forEach((container) => {
+		container.setAttribute('aria-label', `${short} Comments`);
+		container.innerHTML = html;
+	});
+}
+
+/**
+ * @description
+ * Makes an api call to fetch the comment counts for all Cards on the page
+ *
+ * For each Card with comments found (by looking for a marker element) it:
+ *
+ * - Enhances the data, transforming the count from a number to formatted strings
+ * - Creates a html string for each count to be shown using ReactDOM.renderToString
+ * - Mutates the DOM with this new html
+ *
+ * @param {boolean} repeat If true, the fetch call will be repeated on an interval
+ */
+export const FetchCommentCounts = ({ repeat }: Props) => {
+	const [url, setUrl] = useState(buildUrl());
+
+	useApi<{
+		counts: RawCountType[];
+	}>(url, {
+		refreshInterval: repeat ? 15_000 : 0,
+		refreshWhenHidden: false,
+		onSuccess: (data?: { counts?: RawCountType[] }) => {
+			if (data?.counts) {
+				try {
+					const enhancedCounts = enhanceCounts(data.counts);
+					const renderedCounts = renderCounts(enhancedCounts);
+					renderedCounts.forEach((renderedCount) => {
+						insertCount({
+							id: renderedCount.id,
+							html: renderedCount.html,
+							short: renderedCount.short,
+						});
+					});
+				} catch (e) {
+					// Do nothing
+				}
+			}
+			setUrl(buildUrl());
```

Comment:
Why is this set again?

Reply:
Good question. This code wasn't clear but basically we have `url` in state and set it anew on each interval to ensure that we're fetching counts for what's on the page (`Cards` could be adding in dynamically - Onwards, Show More, etc.).

I've added a comment in to explain this and also refactored this code a bit to hopefully make things clearer